central/complianceoperator/v2/report/manager/format/formatter.go

```diff
@@ -102,6 +117,25 @@ func generateRecord(row *report.ResultRow) []string {
 	}
 }
 
+func (f *FormatterImpl) createFailedClusterFileInZip(zipWriter ZipWriter, filename string, failedCluster *storage.ComplianceOperatorReportSnapshotV2_FailedCluster) error {
+	w, err := zipWriter.Create(filename)
+	if err != nil {
+		return err
+	}
+	csvWriter := f.newCSVWriter(failedClusterCSVHeader, true)
+	csvWriter.AddValue(generateFailRecord(failedCluster))
+	return csvWriter.WriteCSV(w)
+}
+
+func generateFailRecord(failedCluster *storage.ComplianceOperatorReportSnapshotV2_FailedCluster) []string {
+	return []string{
```

Comment:
nitpick: it would be good to comment that the order has to match the order defined in `failedClusterCSVHeader`.

Reply:
I added a comment here and also in `genratedRecord`.